packages/vm/src/evm/stack.ts

```diff
@@ -17,14 +16,16 @@ export default class Stack {
     return this._store.length
   }
 
-  push(value: BN) {
-    if (!BN.isBN(value)) {
+  push(value: bigint) {
+    if (typeof value !== 'bigint') {
       throw new VmError(ERROR.INTERNAL_ERROR)
     }
 
-    if (value.gt(MAX_INTEGER)) {
-      throw new VmError(ERROR.OUT_OF_RANGE)
-    }
+    // Why do we have this MAX_INTEGER check here on the BN.js values?
+    // Is this still needed for BigInt? (in doubt I would leave to not chance behavior)
```

Comment:
It's a security/validation check that the evm (or really ethereum overall) should only handle numbers less than MAX_INTEGER

had to check regarding bigint:
> It seems like there is no maximum limit to a BigInt as per spec, which makes sense considering BigInts are supposed to be arbitrary-precision integers, whose "digits of precision are limited only by the available memory of the host system".
https://stackoverflow.com/a/53336729

so it would be wise to keep these checks, but maybe they can be located somewhere where it's checked less frequently so it isn't as taxing on the system (if any extra checks are redundant, not sure)